shared/js/search_provider.js

```diff
@@ -0,0 +1,113 @@
+'use strict';
+
+/* exported SearchProvider */
+/* globals LazyLoader */
+/* globals SettingsListener */
+
+(function(exports) {
+
+  // When the "search_providers.json" file is edited, this version
+  // should be bumped so that existing clients reload the updated
+  // data
+  var VERSION = 1;
+
+  // Cache for current provider configuration
+  var SEARCH_CACHE_KEY = 'search.cache';
+  // The users current provider selection
+  var SEARCH_PROVIDER_KEY = 'search.provider';
+
+  // File containing the provider configurations for all partners
+  var DEFAULT_PROVIDERS_URL = '/shared/js/search_providers.json';
```

Comment:
Still though, it seems the first thing we do is load the JSON file, then write this data to the settings DB. Why don't we just populate the settings value at build time so we don't need to load the file at all during runtime?

I suppose it's just because that's what we used to do previously?